paywall/src/data-iframe/blockchainHandler/keyStatus.js

```diff
@@ -15,13 +15,20 @@ export function isValidKey(key) {
  * @returns {none|submitted|pending|confirming|valid|expired|failed}
  */
 export function getKeyStatus(key, requiredConfirmations) {
-  const valid = isValidKey(key)
+  let valid = isValidKey(key)
   if (!key.transactions || !key.transactions.length) {
     return valid ? 'valid' : 'none'
   }
   // if we reach this line, key.transactions has at least 1 transaction in it
   // the key transactions must be generated by linkTransactionsToKeys in order
   // to be properly sorted
+
+  // For a brief period, a transaction that moves from "submitted" to
+  // "confirming" will have an expiration of 0. We establish a special
+  // case for that here to avoid a bried flicker of the "locked"
+  // paywall state.
+  valid = valid || key.expiration === 0
```

Comment:
Is this related to https://github.com/unlock-protocol/unlock/issues/4377 ?

Reply:
This is strange, because I believe when the user does not have a key unlockjs will actually return an expiration of 0.

Reply:
No, this is a different set of key validation. The one from that issue is located here: https://github.com/unlock-protocol/unlock/blob/949bddfb6fed83064c74d3af16a5396e267a3aaa/paywall/src/utils/validators.js#L165

I agree that it's strange, and in fact that is the behavior of `web3Service`. What happens here is that we have a transaction associated with the key (which is not something that comes from `web3Service`). We haven't actually gotten the information from `web3Service` yet in this case, but we also don't want to treat a key as expired.